base/multidimensional.jl

```diff
@@ -247,18 +247,25 @@ end
 end
 
 # checksize ensures the output array A is the correct size for the given indices
-@noinline throw_checksize_error(arr, dim, idx) = throw(DimensionMismatch("index $d selects $(length(I[d])) elements, but size(A, $d) = $(size(A,d))"))
+@noinline throw_checksize_error(A, dim, idx) = throw(DimensionMismatch("index $dim selects $(length(idx)) elements, but size(A, $dim) = $(size(A,dim))"))
+@noinline throw_checksize_error(A, dim, idx::AbstractArray{Bool}) = throw(DimensionMismatch("index $dim selects $(sum(idx)) elements, but size(A, $dim) = $(size(A,dim))"))
```

Comment:
Oh my mistake, I thought it was checking the size of the input. I should read the comments. Too used to there not being any comments to read, or something.